starlite/datastructures/url.py

```diff
@@ -0,0 +1,105 @@
+from typing import TYPE_CHECKING, Any, NamedTuple, Optional, Union
+from urllib.parse import SplitResult, urlsplit, urlunsplit
+
+if TYPE_CHECKING:
+    from starlite.types import Scope
+
+
+class Address(NamedTuple):
+    host: str
+    port: int
+
+
+def make_absolute_url(url_path: str, base_url: Union[str, "URL"]) -> str:
+    if isinstance(base_url, str):
+        base_url = URL(base_url)
+    netloc = base_url.netloc
+    path = base_url.path.rstrip("/") + url_path
+    return str(URL.from_components(scheme=base_url.scheme, netloc=netloc, path=path))
+
+
+class URL(SplitResult):
+    def __new__(cls, url: Union[str, SplitResult]) -> "URL":
+        if isinstance(url, SplitResult):
+            result = url
+        else:
+            result = urlsplit(url)
+        return super().__new__(cls, *result)
+
+    def __init__(self, url: Union[str, SplitResult]) -> None:
+        pass
```

Comment:
`__new__` is because you cannot simply subclass named tuples and the empty `__init__` is to make typecheckers happy (pyright doesn't seem to care about what's defined in `__new__` in this case).

Reply:
I would rather you add a `pyright: ignore`

Reply:
This would mean adding `pyright: ignore` everywhere the class is being used, since that's what pyright is having issues with.